### PR TITLE
sdrangel: add missing dependencies and fix DAB demodulator

### PR DESCRIPTION
**Pull request by Tungsten842**
###### Description of changes
This fixes the DAB demodulator and the ADS-B map.

Half fix for https://github.com/NixOS/nixpkgs/issues/202721

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by alkeryn — APPROVED** [review]
haven't tested but commit looks clean.

**Comment by Tungsten842** [comment]
Just checking one thing

File: pkgs/applications/radio/sdrangel/default.nix [hunk 1]
```diff
@@ -75,24 +82,31 @@ mkDerivation rec {
     libusb1
     limesuite
     mbelib
+    ninja
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/sdrangel/default.nix:85]: ninja goes to nativeBUildInputs

File: pkgs/applications/radio/sdrangel/default.nix [hunk 2]
```diff
@@ -75,24 +82,31 @@ mkDerivation rec {
...
     opencv3
     qtcharts
+    qtdeclarative
+    qtgraphicaleffects
     qtlocation
     qtmultimedia
+    qtquickcontrols
+    qtquickcontrols2
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/sdrangel/default.nix:93]: You are sure both?

File: pkgs/applications/radio/sdrangel/default.nix (hunk 1 above)
Comment (Tungsten842) [reply pkgs/applications/radio/sdrangel/default.nix]: Fixed it

File: pkgs/applications/radio/sdrangel/default.nix (hunk 2 above)
Comment (Tungsten842) [reply pkgs/applications/radio/sdrangel/default.nix:92]: > It builds without it, so I suspect that the doc is incorrect

It does build without any of the dependencies that I have added, you can probably remove most of the ones that are already specified, but then most decoders/plugins don't work. For example now the ADS-B map doesn't work but it builds fine without the required qt dependency.

**Comment by Tungsten842** [comment]
Result of `nixpkgs-review pr 220746` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sdrangel</li>
  </ul>
</details>

**Pull request merged**